src/style/StyledMain.js

```diff
@@ -1,91 +1,124 @@
 import styled from 'styled-components';
 
 const StyledMain = styled.main`
-  height: 90vh;
-  display: grid;
-  grid-template-areas: 'top' 'bottom';
-  grid-auto-rows: 50vh 40vh;
+  height: 110vh;
+  width: 100vw;
   background-color: #000030;
-  color: #a0a0a0;
+  color: rgba(0, 0, 0, 0.8);
   form {
-    grid-area: top;
+    width: 100vw;
+    height: auto;
+    display: grid;
+    grid-template-areas:
+      'area1'
+      'area2'
+      'area3';
+    grid-template-rows: 5rem 5rem auto;
+    row-gap: 0;
+    place-items: center;
+    scrollbar-width: none;
+  }
+
+  .group1,
+  .group2 {
     display: flex;
     flex-direction: column;
-    align-items: center;
+  }
+  .exercise-row {
+    grid-area: area2;
+    max-height: 2.5rem;
+    width: 80vw;
+    display: flex;
+  }
+
+  .bodypart-row {
+    display: flex;
+    min-height: 4rem;
+    width: 90vw;
+    grid-area: 'area1';
+    justify-content: space-evenly;
     text-align: center;
   }
-  select,
-  .date-input {
-    background-color: #a0a0a0;
+
+  .date-input,
+  select {
+    background-color: #ffefca;
     font-size: 1.1rem;
-    width: 11rem;
-    height: 2.5rem;
-    margin: auto;
+    width: 8rem;
+    height: 2rem;
+    font-weight: 200;
     border-radius: 3px;
+    border: none;
+    margin-right: 3px;
   }
-
-  .bodypart-row {
-    margin: auto;
+  dd {
+    display: flex;
+    align-items: center;
+    white-space: nowrap;
+    padding: 3px;
   }
-  li {
-    list-style: none;
-    width: 240px;
+  .sets-field {
+    margin-left: 1.3rem;
   }
 
   .weeks-plan {
-    grid-area: bottom;
+    height: auto;
     display: flex;
+    align-items: flex-start;
     overflow-x: auto;
+    margin-top: 0;
   }
-
-  .workout-block {
+  .plan-top {
     font-size: 1.1rem;
-    background-color: beige;
-    box-shadow: 1px 1px silver;
+    background-color: #ffefca;
+    width: 90vw;
+    height: auto;
+    grid-area: area3;
     position: relative;
-    min-width: 15rem;
-    min-height: 8rem;
+    text-align: center;
     border-radius: 7px;
-    margin: 0.5rem;
-    display: flex;
-    flex-direction: column;
-    align-items: center;
-  }
-  label {
-    margin-right: 3rem;
-    color: #a0a0a0;
   }
 
-  .plan-top {
-    min-width: 22rem;
-    box-shadow: 1px 1px silver;
-    position: relative;
+  .plan-bottom {
+    background-color: #ffefca;
+    font-size: 1.1rem;
+    width: fit-content;
+    height: fit-content;
+    border-radius: 7px;
+    display: grid;
+    grid-template-areas:
+      'area4'
+      'area5'
+      'area6';
+    grid-template-rows: 2rem 2rem auto;
+    place-items: center;
+    margin: 0 0.5rem;
+  }
+  h4 {
+    grid-area: area4;
+  }
+  h3 {
+    margin-top: 0;
+    grid-area: area5;
   }
+
   .sets-input {
-    position: absolute;
-    right: 0;
     width: 2.2rem;
-    height: 1.1rem;
+    height: 1.2rem;
     border: none;
+    position: absolute;
+    right: 0;
   }
-  button {
-    font-size: 1.1rem;
+
+  .done-button,
+  .set-button {
```

Comment:
**Usage of classes in styled Components**
Many roads lead to Rome. 🙃 
You decided to use classes in combination with styled components and it works. But your styled components become very nested, which can lead to redundant code and bugs become more difficult to track. 

Rather use single Styled Components with `props, css helper`...  here's an [article I found](https://www.smashingmagazine.com/2020/07/styled-components-react/) (quickly) that gives a pretty good overview.